sources/GGTrackerManager.m

```diff
@@ -972,46 +972,61 @@ - (void)startWatchingOrderWithUUID:(NSString *_Nonnull)uuid
-                    
-                    if (!shareUUID) {
-                        shareUUID = [socketResponse objectForKey:@"share_uuid"];
+                    BOOL isShareUUIDExpired = NO;
+                    id expiredObj = [socketResponse objectForKey:@"expired"];
+                    if ([expiredObj isKindOfClass:[NSNumber class]]) {
+                        isShareUUIDExpired = ((NSNumber *)expiredObj).boolValue;
                     }
-
-                    GGSharedLocation *sharedLocation  = [[GGSharedLocation alloc] initWithData:[socketResponse objectForKey:@"shared_location"] ];
-                    
-                    // updated the order model
-                    activeOrder.sharedLocationUUID = shareUUID;
-                    activeOrder.sharedLocation = sharedLocation;
-                    [_liveMonitor addAndUpdateOrder:activeOrder];
                     
-                    if (self.httpManager && shareUUID) {
-                        // try to get the full order object once
-                        [self startRESTWatchingOrderByOrderUUID:uuid sharedUUID:shareUUID withCompletionHandler:^(BOOL success, NSDictionary * _Nullable response, GGOrder * _Nullable order, NSError * _Nullable error) {
-                           
-                            if (success && order) {
-                                order.sharedLocation = sharedLocation;
-                                
-                                [_liveMonitor addAndUpdateOrder:order];
+                    if (isShareUUIDExpired) {
+                        [activeOrder setStatus:OrderStatusDone];
+                        
+                        if ([delegateOfOrder respondsToSelector:@selector(watchOrderSucceedForOrder:)]) {
+                            [delegateOfOrder watchOrderSucceedForOrder:activeOrder];
+                        }
+                    }
+                    else {
+                        NSString *shareUUID = shareduuid;
+                        
+                        if (!shareUUID) {
+                            shareUUID = [socketResponse objectForKey:@"share_uuid"];
+                        }
+                        
+                        GGSharedLocation *sharedLocation  = [[GGSharedLocation alloc] initWithData:[socketResponse objectForKey:@"shared_location"] ];
+                        
+                        // updated the order model
+                        activeOrder.sharedLocationUUID = shareUUID;
+                        activeOrder.sharedLocation = sharedLocation;
+                        [_liveMonitor addAndUpdateOrder:activeOrder];
+                        
+                        if (self.httpManager && shareUUID) {
+                            // try to get the full order object once
+                            [self startRESTWatchingOrderByOrderUUID:uuid sharedUUID:shareUUID withCompletionHandler:^(BOOL success, NSDictionary * _Nullable response, GGOrder * _Nullable order, NSError * _Nullable error) {
```

Comment:
this is the wrong REST method to use. notice that you are try to call the REST watch order instead of just GET on the order. this part should be broken to two use cases.
- when success false and no expiry and you have a valid shareduui you should call the REST watch order method
- in any other case (excluding expired use case you added) you should cal the GET orderByUUID